src/main/java/lotto/domain/LottoGameResult.java

```diff
@@ -0,0 +1,49 @@
+package lotto.domain;
+
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class LottoGameResult {
+
+    private static final int FIFTH_MATCHING_COUNT = 2;
+    private static final double TWO_DECIMAL_PLACES = 100.0;
+    private final List<Rank> ranks;
+    private final PurchaseAmount purchaseAmount;
+
+    public LottoGameResult(List<Rank> ranks, PurchaseAmount purchaseAmount) {
+        this.ranks = ranks;
+        this.purchaseAmount = purchaseAmount;
+    }
+
+    public Map<Rank, Integer> getRanks() {
+        Map<Integer, List<Rank>> collect = ranks.stream()
+                .filter(rank -> rank.getMatchingNumberCount() > FIFTH_MATCHING_COUNT)
+                .collect(Collectors.groupingBy(Rank::getMatchingNumberCount));
+        Map<Rank, Integer> result = init();
+        collect.forEach((key, value) -> result.put(Rank.value(key), result.get(Rank.value(key)) + 1));
+
+        return result;
+    }
```

Comment:
결과를 구하는 로직이 조금 복잡한 것 같아요 😭 현재 구조를 더 간단하게 리팩토링 해볼 수 있을 것 같아요 :)
```suggestion
    public Map<Rank, Integer> getRanks() {
        Map<Rank, Integer> result = init();
        ranks.stream()
                .filter(rank -> result.containsKey(rank))
                .forEach(rank -> {
                    result.put(rank, result.getOrDefault(rank, 0) + 1);
                });
        return result;
    }
```